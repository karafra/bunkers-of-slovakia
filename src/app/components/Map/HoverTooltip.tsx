import { CSSProperties } from 'styled-components';

export interface HoverTooltipProps {
  x: number;
  y: number;
  count?: number;
  name?: string;
  description?: string;
  cluster?: boolean;
}
/*
pointer-events: none; 
position: absolute;
z-index: 9; font-size:
12px; padding: 8px;
background: #000;
color: #fff;
min-width: 160px;
max-height: 240px;
overflow-y: hidden;
*/

const styles: CSSProperties = {
  padding: '10px',
  borderRadius: '5px',
  color: 'black',
  backgroundColor: 'whitesmoke',
  position: 'absolute',
  minWidth: '240px',
  maxWidth: '10%',
  width: 'fit-content',
  blockSize: 'fit-content',
  overflow: 'scroll',
  boxShadow: '0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)',
  overflowX: 'hidden',
  pointerEvents: 'none',
};

const HoverTooltip = (props: HoverTooltipProps) => {
  const cluster = props.cluster || false;
  return cluster ? (
    <div style={{ ...styles, left: props.x, top: props.y }}>
      {props.count} buildings
    </div>
  ) : (
    <div
      className="tooltip"
      style={{
        ...styles,
        left: props.x,
        top: props.y,
      }}
    >
      {props.name}
      {props.description ? ` (${props.description})` : ''}
    </div>
  );
};

export default HoverTooltip;
