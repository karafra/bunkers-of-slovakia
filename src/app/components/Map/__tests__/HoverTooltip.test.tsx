import React from 'react';
import { render } from '@testing-library/react';
import HoverTooltip from '../HoverTooltip';

describe('<HoverTooltip />', () => {
  it('should render tooltip cluster', () => {
    // Given
    const x = 0;
    const y = 0;
    const count = 25;
    // When
    const rendering = render(
      <HoverTooltip cluster x={x} y={y} count={count} />,
    );
    // Then
    expect(rendering).toMatchSnapshot();
    expect(rendering.container.querySelector('div')?.style.left).toBe(`${x}px`);
    expect(rendering.container.querySelector('div')?.style.top).toBe(`${y}px`);
  });

  describe('should render tooltip', () => {
    it('cluster with description', () => {
      // Given
      const x = 0;
      const y = 0;
      const name = 'name';
      const description = 'description';
      // When
      const rendering = render(
        <HoverTooltip x={x} y={y} name={name} description={description} />,
      );
      // Then
      expect(rendering).toMatchSnapshot();
      expect(rendering.container.querySelector('div')?.style.left).toBe(
        `${x}px`,
      );
      expect(rendering.container.querySelector('div')?.style.top).toBe(
        `${y}px`,
      );
      expect(rendering.container.querySelector('div')?.innerHTML).toContain(
        name,
      );
      expect(rendering.container.querySelector('div')?.innerHTML).toContain(
        description,
      );
    });
    it('cluster without description', () => {
      // Given
      const x = 0;
      const y = 0;
      const name = 'name';
      // When
      const rendering = render(<HoverTooltip x={x} y={y} name={name} />);
      // Then
      expect(rendering).toMatchSnapshot();
      expect(rendering.container.querySelector('div')?.style.left).toBe(
        `${x}px`,
      );
      expect(rendering.container.querySelector('div')?.style.top).toBe(
        `${y}px`,
      );
      expect(rendering.container.querySelector('div')?.innerHTML).toContain(
        name,
      );
      expect(rendering.container.querySelector('div')?.innerHTML).toContain('');
    });
  });
});
