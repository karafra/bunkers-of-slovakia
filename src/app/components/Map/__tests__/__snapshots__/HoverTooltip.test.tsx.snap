// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HoverTooltip /> should render tooltip cluster 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  padding: '10px',border-radius:'5px',color:'black',background-color:'whitesmoke',position:'absolute',min-width:'240px',max-width:'10%',width:'fit-content',block-size:'fit-content',overflow:'scroll',box-shadow:'0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)',overflow-x:'hidden',pointer-events:'none',;
}

<body>
    <div>
      <div
        class="c0 tooltip"
        style="left: 0px; top: 0px;"
      >
        25 buildings
      </div>
    </div>
  </body>,
  "container": .c0 {
  padding: '10px',border-radius:'5px',color:'black',background-color:'whitesmoke',position:'absolute',min-width:'240px',max-width:'10%',width:'fit-content',block-size:'fit-content',overflow:'scroll',box-shadow:'0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)',overflow-x:'hidden',pointer-events:'none',;
}

<div>
    <div
      class="c0 tooltip"
      style="left: 0px; top: 0px;"
    >
      25 buildings
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<HoverTooltip /> should render tooltip cluster with description 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  padding: '10px',border-radius:'5px',color:'black',background-color:'whitesmoke',position:'absolute',min-width:'240px',max-width:'10%',width:'fit-content',block-size:'fit-content',overflow:'scroll',box-shadow:'0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)',overflow-x:'hidden',pointer-events:'none',;
}

<body>
    <div>
      <div
        class="c0 tooltip"
        style="left: 0px; top: 0px;"
      >
        <div>
          name
           (description)
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  padding: '10px',border-radius:'5px',color:'black',background-color:'whitesmoke',position:'absolute',min-width:'240px',max-width:'10%',width:'fit-content',block-size:'fit-content',overflow:'scroll',box-shadow:'0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)',overflow-x:'hidden',pointer-events:'none',;
}

<div>
    <div
      class="c0 tooltip"
      style="left: 0px; top: 0px;"
    >
      <div>
        name
         (description)
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<HoverTooltip /> should render tooltip cluster without description 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  padding: '10px',border-radius:'5px',color:'black',background-color:'whitesmoke',position:'absolute',min-width:'240px',max-width:'10%',width:'fit-content',block-size:'fit-content',overflow:'scroll',box-shadow:'0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)',overflow-x:'hidden',pointer-events:'none',;
}

<body>
    <div>
      <div
        class="c0 tooltip"
        style="left: 0px; top: 0px;"
      >
        <div>
          name
          
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  padding: '10px',border-radius:'5px',color:'black',background-color:'whitesmoke',position:'absolute',min-width:'240px',max-width:'10%',width:'fit-content',block-size:'fit-content',overflow:'scroll',box-shadow:'0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22)',overflow-x:'hidden',pointer-events:'none',;
}

<div>
    <div
      class="c0 tooltip"
      style="left: 0px; top: 0px;"
    >
      <div>
        name
        
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
